src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -645,44 +580,49 @@ FILE* Compiler::fgOpenFlowGraphFile(bool* wbDontClose, Phases phase, LPCWSTR typ
-//        will produce a Foo.svg file that can be opened with any svg-capable browser (e.g. IE).
+//      - The command:
+//           "C:\Program Files (x86)\Graphviz2.38\bin\dot.exe" -Tsvg -oFoo.svg -Kdot Foo.dot
+//        will produce a Foo.svg file that can be opened with any svg-capable browser.
+//    - https://sketchviz.com/
 //    - http://rise4fun.com/Agl/
 //      - Cut and paste the graph from your .dot file, replacing the digraph on the page, and then click the play
 //        button.
 //      - It will show a rotating '/' and then render the graph in the browser.
-//    MSAGL has also been open-sourced to https://github.com/Microsoft/automatic-graph-layout.git.
+//    MSAGL has also been open-sourced to https://github.com/Microsoft/automatic-graph-layout.
 //
 //    Here are the config values that control it:
 //      COMPlus_JitDumpFg       A string (ala the COMPlus_JitDump string) indicating what methods to dump flowgraphs
 //                              for.
 //      COMPlus_JitDumpFgDir    A path to a directory into which the flowgraphs will be dumped.
 //      COMPlus_JitDumpFgFile   The filename to use. The default is "default.[xml|dot]".
 //                              Note that the new graphs will be appended to this file if it already exists.
+//      COMPlus_NgenDumpFg      Same as COMPlus_JitDumpFg, but for ngen compiles.
+//      COMPlus_NgenDumpFgDir   Same as COMPlus_JitDumpFgDir, but for ngen compiles.
+//      COMPlus_NgenDumpFgFile  Same as COMPlus_JitDumpFgFile, but for ngen compiles.
 //      COMPlus_JitDumpFgPhase  Phase(s) after which to dump the flowgraph.
 //                              Set to the short name of a phase to see the flowgraph after that phase.
 //                              Leave unset to dump after COLD-BLK (determine first cold block) or set to * for all
 //                              phases.
-//      COMPlus_JitDumpFgDot    Set to non-zero to emit Dot instead of Xml Flowgraph dump. (Default is xml format.)
+//      COMPlus_JitDumpFgDot    0 for xml format, non-zero for dot format. (Default is dot format.)
 
 bool Compiler::fgDumpFlowGraph(Phases phase)
 {
     bool result        = false;
     bool dontClose     = false;
-    bool createDotFile = false;
-    if (JitConfig.JitDumpFgDot())
-    {
-        createDotFile = true;
-    }
+    bool createDotFile = true;
 
-    FILE* fgxFile = fgOpenFlowGraphFile(&dontClose, phase, createDotFile ? W("dot") : W("fgx"));
+#ifdef DEBUG
+    createDotFile = JitConfig.JitDumpFgDot() != 0;
+#endif // DEBUG
 
+    FILE* fgxFile = fgOpenFlowGraphFile(&dontClose, phase, createDotFile ? W("dot") : W("fgx"));
     if (fgxFile == nullptr)
     {
         return false;
     }
+
     bool        validWeights  = fgHaveValidEdgeWeights;
-    double      weightDivisor = (double)fgCalledCount;
+    double      weightDivisor = (double)(fgCalledCount == BB_ZERO_WEIGHT ? 1.0 : fgCalledCount); // avoid divide by zero
```

Comment:
```suggestion
    double      weightDivisor = (double)fgFirstBB->getCalledCount(this);
```
but maybe make that method static and / or move off of BasicBlock.